integration-tests/tests/src/tests/mixed.ts

```diff
@@ -87,7 +115,71 @@ const MixedNullableSchema: Realm.ObjectSchema = {
 
 const MixedSchema: Realm.ObjectSchema = {
   name: "MixedClass",
-  properties: { value: "mixed" },
+  properties: {
+    value: "mixed",
+  },
+};
+
+const MixedAndEmbeddedSchema: Realm.ObjectSchema = {
+  name: "MixedAndEmbedded",
+  properties: {
+    mixedValue: "mixed",
+    embeddedObject: "EmbeddedObject?",
+  },
+};
+
+const EmbeddedObjectSchema: Realm.ObjectSchema = {
+  name: "EmbeddedObject",
+  embedded: true,
+  properties: {
+    value: "mixed",
+  },
+};
+
+const CollectionsOfMixedSchema: Realm.ObjectSchema = {
+  name: "CollectionsOfMixed",
+  properties: {
+    list: "mixed[]",
+    dictionary: "mixed{}",
+    set: "mixed<>",
+  },
+};
+
+const bool = true;
+const int = 123;
+const double = 123.456;
+const d128 = BSON.Decimal128.fromString("6.022e23");
+const string = "hello";
+const date = new Date();
+const oid = new BSON.ObjectId();
+const uuid = new BSON.UUID();
+const nullValue = null;
+const uint8Values = [0, 1, 2, 4, 8];
+const uint8Buffer = new Uint8Array(uint8Values).buffer;
+const unmanagedRealmObject: IMixedSchema = { value: 1 };
+
+// The `unmanagedRealmObject` is not added to these collections since a managed
+// Realm object will be added by the individual tests after one has been created.
+const flatListAllTypes: unknown[] = [bool, int, double, d128, string, date, oid, uuid, nullValue, uint8Buffer];
+const flatDictionaryAllTypes: Record<string, unknown> = {
+  bool,
+  int,
+  double,
+  d128,
+  string,
+  date,
+  oid,
+  uuid,
+  nullValue,
+  uint8Buffer,
+};
```

Comment:
I like that you're defining these const values instead of having them inlined in the tests 👍